### ISSUE TITLE
Exception: Destination array was not long enough

### ISSUE THREAD

**Issue by citron4000** [Placeholder-Software/Dissonance#16]
Hello,

I have updated to 0.0.8 and testing the Unity demo scene and two Oculus Rift on LAN.

Question:

When I'm using Push To Talk option and continuously the key to talk, I can talk for about 30 seconds and then no voice is transmitted anymore. I need to release the key and then it works again. Is it the intented behaviour?

Exception:

Because VAD is under development, I'm quickly modifying the source code and adding `OpenMic` in `CommActivationMode` and the following code in `VoiceBroadcastTrigger`:

```
                case CommActivationMode.OpenMic:
                    activate = true;
                    break;
```

After a variable amount of time (somtimes before 30 seconds), I get the following exception:

![dissonance-0 0 8](https://cloud.githubusercontent.com/assets/6678654/21042887/3a46acac-be2f-11e6-9b5f-62cc9d919983.PNG)

Then no voice is transmitted anymore.

Any idea why?

Thanks!

**Comment by martindevans** [comment]
> modifying the source code and adding OpenMic

That's perfectly fine. This is exactly how the original OpenMic setting worked so this shouldn't introduce any bugs.

> I can talk for about 30 seconds and then no voice is transmitted anymore. I need to release the key and then it works again. Is it the intended behaviour?

This is definitely not the intended behaviour! PTT should keep transmitting for as long as you keep holding it. Just to clarify is the transmission cutting out associated with this exception happening?

I just tried talking continuously for about 90 seconds and can't reproduce this issue. However, from looking at that stack trace I have a guess what the problem could be, I'll get back to you soon :)

**Comment by martindevans** [comment]
Okay a progress update:

I found a bug which was probably causing the exception you were suffering from (a bit of sanity checking wasn't fully sane). That's fixed now (as usual, available in a day or two with 0.0.9).

However, ultimately the the bug was triggered by trying to write too much into a buffer. So the real question is why is too much being written into a buffer? It should be draining out samples almost as soon as they arrive. The only way I can see this happening is if the mic delivers a large amount of audio in one go (more than 2 frames). Could you try seeing how much is getting written into the buffer when this exception happens? Substitute line 265 of MicrophoneCapture:


Current:
```
_rawMicSamples.Write(buffer, 0, buffer.Length);
```

Replacement:

```
try
{
  _rawMicSamples.Write(buffer, 0, buffer.Length);
}
catch (Exception e)
{
  Log.Error("Boom! Writing {0} samples into the buffer. There are already {1} unread samples in the buffer", buffer.Length, _rawMicSamples.Count);
}
```

**Comment by citron4000** [comment]
> This is definitely not the intended behaviour! PTT should keep transmitting for as long as you keep holding it. Just to clarify is the transmission cutting out associated with this exception happening?

Actually no, this Exceptions occur only with OpenMic on. For PPT, I don't see any errors when it stops transmitting :/

> Could you try seeing how much is getting written into the buffer when this exception happens? Substitute line 265 of MicrophoneCapture:

Sure, I will send it on Monday from the office.

**Comment by martindevans** [comment]
I just had another poke around at this.First I tried adding in your OpenMic code and still couldn't reproduce it (so that's not the problem). Second I remembered you're using different quality settings (Small frame, high quality) and I managed to reproduce this!

So, to mitigate this return to the default frame size setting (Medium). I'll get back to you again once I have investigated the root cause.
